src/Data/Avro/Encode.hs

```diff
@@ -72,81 +73,88 @@ newSyncBytes = BL.pack . DL.take 16 . randoms <$> initTFGen
 
 -- |Encode chunks of objects into a container, using 16 random bytes for
 -- the synchronization markers.
-encodeContainer :: EncodeAvro a => Schema -> [[a]] -> IO BL.ByteString
-encodeContainer sch xss =
+encodeContainer :: EncodeAvro a => Codec -> Schema -> [[a]] -> IO BL.ByteString
```

Comment:
consider calling this `encodeContainerWithCodec`, to let `encodeContainer` keep its old type and avoid API change?

Reply:
I've considered this but this pattern seems to not scale very well. There is `encodeContainer`, `encodeContainerWithSync`, `encodeContainerWithCodec` and `encodeContainerWithSyncAndCodec`. I kept compatibility in `Data.Avro`s `encodeContainer` though.